ethereumj-core/src/main/java/org/ethereum/sharding/proposer/ProposerServiceImpl.java

```diff
@@ -68,17 +75,25 @@ public ProposerServiceImpl(BeaconProposer proposer, BeaconChain beaconChain,
     }
 
     @Override
-    public void init(BeaconState state) {
+    public void init(BeaconState state, byte[]... pubKeys) {
+        assert pubKeys.length > 0;
+
+        this.validatorIndices = new HashSet<>();
+
+        for (byte[] pubKey : pubKeys) {
+            Validator validator = state.getValidatorSet().getByPubKey(pubKey);
+            if (validator != null) {
+                this.validatorIndices.add(validator.getIndex());
+                this.pubKeysMap.put(validator.getIndex(), validator.getPubKey());
+            } else {
+                // something went wrong
+                logger.error("Failed to start proposer for {}: validator does not exist", HashUtil.shortHash(pubKey));
+                return;
```

Comment:
I'd either throw an exception on this or continue to parse everything the rest. Both would be better, I guess.

Reply:
Agree on that! It's too silent